src/librustc/ty/context.rs

```diff
@@ -482,6 +482,23 @@ impl<'tcx> TypeckTables<'tcx> {
         }
     }
 
+    pub fn opt_type_dependent_def(&self, id: HirId) -> Option<&Def> {
+        validate_hir_id_for_typeck_tables(self.local_id_root, id, false);
+        self.type_dependent_defs.get(&id.local_id)
+    }
+
+    pub fn opt_type_dependent_def_id(&self, id: HirId) -> Option<DefId> {
+        match self.opt_type_dependent_def(id) {
```

Comment:
Remove this `match` by using the `map` method on the result of `opt_type_dependent_def`